apis/supernova/src/handler/data-property-handler.js

```diff
@@ -113,72 +270,172 @@ class DataPropertyHandler {
     return this.addDimensions(dimensions, true);
   }
 
+  /**
+   * Adds multiple alternative field dimensions to the handler.
+   * @param {FieldDimensions[]} args - Array of field dimension args.
+   * @returns {Promise<qix.NxDimension[]>} The result of addDimensions.
+   * @description Creates and adds multiple alternative field dimensions.
+   * @memberof DataPropertyHandler
+   * @example
+   * await handler.addAltFieldDimensions([{ field: 'A' }, { field: 'B' }]);
+   */
   async addAltFieldDimensions(args) {
     const dimensions = args.map(({ field }) => this.createFieldDimension(field));
     return this.addDimensions(dimensions, true);
   }
 
+  /**
+   * Adds an alternative field dimension to the handler.
+   * @param {string} field - The field definition for the dimension.
+   * @param {string=} label - The field label for the dimension.
+   * @param {qix.NxDimension=} defaults - Default properties for the dimension.
+   * @returns {Promise<qix.NxDimension=>} The result of addDimension.
+   * @description Creates and adds an alternative field dimension.
+   * @memberof DataPropertyHandler
+   * @example
+   * handler.addAlternativeFieldDimension('A', 'AA');
+   */
   addAlternativeFieldDimension(field, label, defaults) {
     const dimension = this.createFieldDimension(field, label, defaults);
     return this.addDimension(dimension, true);
   }
 
+  /**
+   * Adds an alternative library dimension to the handler.
+   * @param {string} id - The library dimension id.
+   * @param {qix.NxDimension=} defaults - Default properties for the dimension.
+   * @returns {Promise<qix.NxDimension=>} The result of addDimension.
+   * @description Creates and adds an alternative library dimension.
+   * @memberof DataPropertyHandler
+   * @example
+   * handler.addAlternativeLibraryDimension([{ id: 'A' }, { id: 'B', defaults: { ... } }]);
+   */
   addAlternativeLibraryDimension(id, defaults) {
     const dimension = this.createLibraryDimension(id, defaults);
     return this.addDimension(dimension, true);
   }
 
+  /**
+   * Gets the minimum number of dimensions allowed.
+   * @returns {number} The minimum number of dimensions.
+   * @description Returns the minimum number of dimensions allowed by the handler.
+   * @memberof DataPropertyHandler
+   * @example
+   * const min = handler.minDimensions();
+   */
   minDimensions() {
     if (typeof this.dimensionDefinition.min === 'function') {
       return this.dimensionDefinition.min.call(null, this.properties, this);
     }
-
     return this.dimensionDefinition.min || 0;
   }
 
+  /**
+   * Gets the maximum number of dimensions allowed.
+   * @param {number} [decrement=0] - The number to decrement from the current number of dimensions.
+   * @returns {number} The maximum number of dimensions allowed.
+   * @description Checks if the max property is a function and calls it with the current number of dimensions, or returns a default value.
```

Comment:
the argument to the max dimensions function is current number of **measures** (decremented by decrement parameter)

```suggestion
   * Gets the maximum number of dimensions allowed.
   * @param {number} [decrement=0] - The number to decrement from the current number of measures.
   * @returns {number} The maximum number of dimensions allowed.
   * @description Checks if the max property is a function and calls it with the current number of measures, or returns a default value.
```